packages/desktop-client/src/components/transactions/MobileTransaction.js

```diff
@@ -99,6 +175,768 @@ function Status({ status }) {
+  //   });
+  // };
+
+  // onAddSplit = () => {
+  //   this.props.navigation.navigate('CategorySelect', {
+  //     title: 'Select a category',
+  //     onSelect: categoryId => {
+  //       let transactions = this.state.transactions;
+
+  //       // Split the transaction
+  //       let { data } = addSplitTransaction(transactions, transactions[0].id);
+  //       // Set the initial category
+  //       data[data.length - 1].category = categoryId;
+
+  //       this.setState({ transactions: data }, this.focusSplit);
+  //     },
+  //   });
+  // };
+
+  // focusSplit = () => {
+  //   if (this.lastChildAmount) {
+  //     this.lastChildAmount.focus();
+  //   }
+  // };
+
+  // onDeleteSplit = transaction => {
+  //   let { transactions } = this.state;
+  //   let { data } = deleteTransaction(transactions, transaction.id);
+  //   this.setState({ transactions: data });
+  // };
+
+  // TODO: `renderActions` was used to render swipe actions on split transaction
+  // rows. It enabled you to swipe, revealing a delete action, and if you swiped
+  // all the way it would perform the delete action. The entire swipe was
+  // animated. Swipe actions are a native paradigm and we'll need a web-friendly
+  // replacement.
+  //
+  // renderActions = (progress, dragX) => {
+  //   const trans = dragX.interpolate({
+  //     inputRange: [-101, -100, -50, 0],
+  //     outputRange: [-6, -5, -5, 20],
+  //   });
+  //   return (
+  //     <RectButton
+  //       onPress={this.close}
+  //       style={{
+  //         flex: 1,
+  //         justifyContent: 'center',
+  //         backgroundColor: colors.r4,
+  //       }}
+  //     >
+  //       <Animated.Text
+  //         style={{
+  //           color: 'white',
+  //           textAlign: 'right',
+  //           transform: [{ translateX: trans }],
+  //         }}
+  //       >
+  //         Delete
+  //       </Animated.Text>
+  //     </RectButton>
+  //   );
+  // };
+
+  render() {
+    const {
+      adding,
+      categories,
+      accounts,
+      payees,
+      renderChildEdit,
+      navigation,
+      onDelete,
+    } = this.props;
+    const { editingChild } = this.state;
+    const transactions = this.serializeTransactions(
+      this.state.transactions || [],
+    );
+    const [transaction, ..._childTransactions] = transactions;
+    const { payee: payeeId, category, account: accountId } = transaction;
+
+    // Child transactions should always default to the signage
+    // of the parent transaction
+    let forcedSign = transaction.amount < 0 ? 'negative' : 'positive';
+
+    let account = getAccountsById(accounts)[accountId];
+    let payee = payees && payeeId && getPayeesById(payees)[payeeId];
+    let transferAcct =
+      payee &&
+      payee.transfer_acct &&
+      getAccountsById(accounts)[payee.transfer_acct];
+
+    let descriptionPretty = getDescriptionPretty(
+      transaction,
+      payee,
+      transferAcct,
+    );
+
+    return (
+      // <KeyboardAvoidingView>
```

Comment:
It would be nice to retain the keyboard avoidance behavior, not sure how to do that on the web though — possibly for a future PR if it isn’t easy?

Reply:
I agree, it would be nice to keep this behavior around. It seems `react-native-web` contains a `KeyboardAvoidingView` impl: https://github.com/necolas/react-native-web/blob/15f78993a58202b4b9858943a799c152e3067385/packages/react-native-web/src/exports/KeyboardAvoidingView/index.js

When I get the chance I'll try swapping that in and seeing what happens. If it ends up being too much work I would say let's definitely leave this for a follow-up.

Reply:
Gave the code there a look, it seems like it doesn’t actually do anything so probably fine to remove and re-evaluate if there are complaints about the keyboard behavior.

Reply:
Ah, unfortunate. In generally as far as I know "keyboard avoiding view" isn't really a thing on the web, so yeah, not sure what the feasibility of even implementing this would be.